### DISCUSSION TITLE
Cannot resolve client hostnames using Reverse DNS, PTR queries with IPv6

### DISCUSSION THREAD

**Discussion by samstringerhye** [AdguardTeam/AdGuardHome#6445] [Q&A]
So I have, I believe, set this up properly and I am able to resolve IPv4 hostnames of clients properly, but I am unable to resolved hostnames of IPv6 clients. Is there any trick to properly setting this up? 

These are my current DNS Settings:

**Upstream DNS servers**
`quic://dns.nextdns.io
https://cloudflare-dns.com/dns-query
https://dns.quad9.net/dns-query
[/in-addr.arpa/]192.168.10.1 2600:4040:79a5:9300::1
[/ip6.arpa/]192.168.10.1 2600:4040:79a5:9300::1
[/localdomain/]192.168.10.1 2600:4040:79a5:9300::1
[/local/]192.168.10.1 2600:4040:79a5:9300::1`

**Bootstrap DNS Servers**
`1.1.1.1
1.0.0.1
2606:4700:4700::1111
2606:4700:4700::1001`

**Private reverse DNS Servers**
`192.168.10.1
2600:4040:79a5:9300::1`

I'm hoping that I have the right settings

**Comment by MNLierman** [comment]
You received no comments, yet Google has shown you at the top of the search results. I love how Google and Bing randomly pick things from around the Internet that are, subjectively, older and are otherwise random in regard to relivency. There is nothing making your Dicussion thread more relevant now than it was when you posted it. Judging by the lack of commnets, I would say that the added traffic to your comment (I keep seeing it when searching for AdGuardHome IPv6 hostname lookup issues – in various wording of the queries. 

All that said, did you find a solution? I'm leaning to believe that AGH is indeed sending the IPv6 queries to the IP I tell it to, but it doesn't respond with client names. The IP AGH is sending to for me is my router with dnsmasq. I don't know that dnsmasq specifically has a log of all IPv6 clients and names. This is an NDP (Neighbor Discovery Protocol) subject, and AGH really should have an option to use ip neight and resovle the hostnames itself. For each IPv6 address that it doesn't recognize, it should send an NDP message to the client and ask for it's hostname. I'll try to comment back if I find the answer.
